src/OpenTelemetry.AutoInstrumentation/Instrumentations/GraphQL/GraphQLCommon.cs

```diff
@@ -58,7 +58,10 @@ internal static Activity CreateActivityFromExecuteAsync(IExecutionContext execut
                 tags.Document = query;
             }
 
-            activity = ActivitySource.StartActivityWithTags(operation, ActivityKind.Server, tags);
+            var activityCurrent = Activity.Current;
+            var activityKind = activityCurrent != null && activityCurrent.Recorded ? ActivityKind.Internal : ActivityKind.Server;
```

Comment:
Shouldn't we also check if `activityCurrent.Kind` is `ActivityKind.Server` or `ActivityKind.Internal`? This covers only if the ASP.NET spans are the direct parents. What if there is some other instrumentation as parent? E.g. some in-proc HttpClient calling GraphQL with ASP.NET instrumentation disabled.

Reply:
Ref.: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind

I will check it. Especially scenarios, were we have a remote parent.